resolwe_bio/processes/reads_processing/bbduk.yml

```diff
@@ -11,221 +11,363 @@
-            A barcode must be the last part of the read header.
+        - name: barcode_filter
+          label: >
+            Remove reads with unexpected barcodes if barcodes are set, or barcodes containing 'N'
+            otherwise
+          type: basic:boolean
+          default: false
+          description: A barcode must be the last part of the read header.
         - name: barcode_files
-          label: List of files of barcodes
+          label: Barcode sequences
           type: list:data:seq:nucleotide
           required: false
-          description: >
-            Comma-delimited list of files of barcodes.
-        - name: barcode_seq
-          label: List of barcodes.
+        - name: barcode_sequences
+          label: Literal barcode sequences
           type: list:basic:string
           required: false
+          default: []
+        - name: x_min
+          label: Minimum X coordinate
+          type: basic:integer
+          default: -1
+        - name: y_min
+          label: Minimum Y coordinate
+          type: basic:integer
+          default: -1
+        - name: x_max
+          label: Maximum X coordinate
+          type: basic:integer
+          default: -1
+          description: Used only if positive.
+        - name: y_max
+          label: Maximum Y coordinate
+          type: basic:integer
+          default: -1
+          description: Used only if positive.
+    - name: complexity
+      label: Complexity parameters
+      hidden: '!show_advanced'
+      group:
+        - name: entropy
+          label: Minimum entropy
+          type: basic:decimal
+          default: -1.0
           description: >
-            Comma-delimited list of barcodes.
-    - name: threads
-      label: Number of threads
-      type: basic:integer
-      required: false
-      description: >
-        Set number of threads to use; default is number of logical processors.
+            Set between 0 and 1 to filter reads with entropy below that value. Higher is more
+            stringent.
+        - name: entropy_window
+          label: Length of sliding window used to calculate entropy
+          type: basic:integer
+          default: 50
+        - name: entropy_k
+          label: Length of kmers used to calcuate entropy
+          type: basic:integer
+          default: 5
+        - name: min_base_frequency
+          label: Minimum base frequency
+          type: basic:integer
+          default: 0
+        - name: entropy_mask
+          label: Mask low-entropy parts of sequences with N instead of discarding
+          type: basic:boolean
+          default: false
   output:
     - name: fastq
-      label: Reads file
+      label: Remaining reads
+      type: list:basic:file
+    - name: dumped
+      label: Dumped reads
+      type: list:basic:file
+    - name: statistics
+      label: Statistics
+      type: list:basic:file
+    - name: kmers
```

Comment:
What is contained in this output files?

Reply:
A list of matching _k_-mer sequences. I removed this output along with dumped reads because I don't see enough value for a typical user. Additionally, the file in `kmers` is much larger than all other outputs combined.